test/new_relic/gem_notifications_tests.rb

```diff
@@ -86,4 +87,20 @@ def test_interpolate_rubygems_url
     gem_name = "velociraptor"
     assert_kind_of String, interpolate_rubygems_url(gem_name)
   end
+
+  def test_cve_bot_message
+    assert_raises(ArgumentError) { cve_bot_message() }
+    assert_raises(ArgumentError) { cve_bot_message("allosaurus") }
+    message = cve_bot_message("allosaurus", "dinotracker.com")
+    assert_equal message, ":rotating_light: allosaurus\n<dinotracker.com|More info here>"
+    assert_kind_of String, message
+  end
```

Comment:
We haven't yet introduced the MiniTest specific RuboCop rules to this project and the existing test files are filled with blocks like this one, so no need to change anything at this time.

But to be thinking of patterns for us to standardize on a team...

Some unit testing style rules forbid the use of multiple expectations / assertions within a single test. For us - especially when the mocking or stubbing is particularly tricky to set up - it has been helpful to use multiple assertions to validate a response. For example, you might perform a web request and want to validate both the HTTP response code AND the response body.

My take that I would like input on from the team would be to only use multiple assertions for the exact same testing scenario.

Looking at `test_cve_bot_message`, I see fully 3 separate scenarios:

- What happens if zero arguments are given?
- What happens if the wrong number of arguments are given?
- What happens if the happy path version of arguments are given?

I'd use 3 separate `test_*` methods to cover these scenarios, even if the first 2 just end up being one liners.

For the last test, using both the `assert_equal` and `assert_kind_of` checks in the same `test_*` method would be ok, because both are validating the same output of the same scenario.

Reply:
I've been thinking the same thing, @fallwith. I agree with your take and am all for adding this to our Dev Team Best Practices.

Reply:
Thank you! This is quite helpful and a best practice to take moving forward :) I've refactored the current and previous tests through [this commit](https://github.com/newrelic/newrelic-ruby-agent/pull/1292/commits/4052615af1e88f7ead3746b7526c03cdc43c73e6).